File: fix-logo.py
from PIL import Image, ImageOps
import os

logo_path = './public/mario.png'

print('Membaca gambar logo...')
img = Image.open(logo_path)
print(f'Dimensi original: {img.size[0]}x{img.size[1]}')

# Convert ke RGB jika perlu
if img.mode == 'RGBA':
    img = img.convert('RGB')

# Crop white borders lebih agresif
print('Menghapus white borders...')
img_trimmed = ImageOps.crop(img, border=200)
print(f'Dimensi setelah crop: {img_trimmed.size[0]}x{img_trimmed.size[1]}')

# Resize ke ukuran square yang lebih kecil (standar printer thermal)
# 384px = lebar penuh printer (32 chars * 12 pixel), logo 200px bagus
target_size = 200
print(f'Resize ke {target_size}x{target_size}...')

# Buat square canvas dengan white background
final_img = Image.new('RGB', (target_size, target_size), 'white')

# Resize logo dengan maintain aspect ratio
img_trimmed.thumbnail((target_size - 20, target_size - 20), Image.Resampling.LANCZOS)

# Paste centered
offset = ((target_size - img_trimmed.width) // 2,
          (target_size - img_trimmed.height) // 2)
final_img.paste(img_trimmed, offset)

# Save
final_img.save(logo_path, 'PNG')
print('Logo berhasil diperbaiki!')
print(f'File: {logo_path}')
print(f'Ukuran final: {target_size}x{target_size}')

File: fix-logo.js
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

const logoPath = path.resolve('./public/mario.png');
const tempPath = path.resolve('./public/mario.temp.png');

async function fixLogoCenter() {
  try {
    console.log('📦 Membaca gambar logo...');

    // Baca metadata gambar original
    const metadata = await sharp(logoPath).metadata();
    console.log(`Dimensi original: ${metadata.width}x${metadata.height}`);

    // Trim ruang kosong (crop whitespace)
    console.log('✂️ Menghapus ruang kosong...');
    const trimmed = await sharp(logoPath)
      .trim({
        background: '#ffffff',
        lineArtTolerance: 0
      })
      .toBuffer({ resolveWithObject: true });

    const trimmedMetadata = trimmed.info;
    console.log(`Dimensi setelah trim: ${trimmedMetadata.width}x${trimmedMetadata.height}`);

    // Tentukan ukuran square (gunakan dimensi yang lebih besar)
    const size = Math.max(trimmedMetadata.width, trimmedMetadata.height) + 50; // tambah padding
    console.log(`Ukuran square target: ${size}x${size}`);

    // Buat gambar baru dengan background putih dan logo centered
    console.log('🎨 Processing gambar agar centered...');

    await sharp(trimmed.data)
      .resize(size, size, {
        fit: 'contain',
        background: { r: 255, g: 255, b: 255, alpha: 1 }
      })
      .png()
      .toFile(tempPath);

    // Ganti file original dengan file baru
    fs.renameSync(tempPath, logoPath);

    console.log('✅ Logo berhasil diperbaiki dan disimpan!');
    console.log(`📍 File: ${logoPath}`);
    console.log(`📐 Ukuran final: ${size}x${size}`);

  } catch (error) {
    console.error('❌ Error:', error.message);
    console.error(error.stack);
    // Clean up temp file jika ada error
    if (fs.existsSync(tempPath)) {
      fs.unlinkSync(tempPath);
    }
    process.exit(1);
  }
}

fixLogoCenter();
